### PR TITLE
WIP edit initiative

### PR DESCRIPTION
**Pull request by Qianfinland**
Now i add feature for editing initiaitve(title and description), i am not that clear whether i need to edit the issue title as well, as you have specific issue on "Edit issue". Any suggestion?

### CODE REVIEW

File: app/controllers/initiative.js [hunk 1]
```diff
@@ -1,22 +0,0 @@
-import Ember from 'ember';
```
Comment (Qianfinland) [inline app/controllers/initiative.js:1]: we didn't use this file in our app, so i delete it

File: tests/acceptance/editing-initiative-test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    Ember.run(App, 'destroy');
+  }
+});
+
+test('Successfully', function(){
+  visit('/initiatives/1').then(function() {
+    click( $("a:contains('Edit initiative')") ).then(function() {
+      expect(find('.title_name').text()).to.equal('Initiative title');
+      fillIn('div.title_value input', "Public health clinic");
+      //I could put any text in the fillIn part,
```
Comment (Qianfinland) [inline tests/acceptance/editing-initiative-test.js]: i don't know how to test value in the in place editor, need some suggestion :)

File: tests/acceptance/editing-initiative-test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+});
+
+test('Update edit', function(){
+  visit('/initiatives/1').then(function() {
+    click( $("a:contains('Edit initiative')") ).then(function() {
+      click( $("button:contains('Update')") ).then(function() {
+        expect(currentURL()).to.equal('/initiatives/1');
+        expect( $("a:contains('Public health clinic')").length ).to.equal(0);
+        expect( $("a:contains('Allocate compensation money to create a local public health clinic')").length ).to.equal(0);
+        // the updated content cannot be predicted, i only test the content is changed, i don't know how to test the unpredictable content
```
Comment (Qianfinland) [inline tests/acceptance/editing-initiative-test.js]: not sure how to test the updated content, i test the validation of the updation. Do you have any comment about this?

File: tests/acceptance/editing-initiative-test.js (hunk 2 above)
Comment (oliverbarnes) [inline tests/acceptance/editing-initiative-test.js]: you don't need to test the value in the editor - you're filling it in, that's enough. what needs to be tested is the outcome after submitting the form

File: tests/acceptance/editing-initiative-test.js (hunk 3 above)
Comment (oliverbarnes) [inline tests/acceptance/editing-initiative-test.js]: which user scenario is this trying to test, exactly?

File: tests/acceptance/editing-initiative-test.js (hunk 2 above)
Comment (Qianfinland) [inline tests/acceptance/editing-initiative-test.js]: before i think that when users want to edit exisitng initiative, after they click the edit button, a box with the previous initiative title and textarea with previous initiative description will appear. So i try to test when the box and textarea appear, they have previous content of initiative.
So i just need to write code as below?
fillIn('div.title_value input');

File: tests/acceptance/editing-initiative-test.js (hunk 3 above)
Comment (Qianfinland) [inline tests/acceptance/editing-initiative-test.js]: i try to test after users change the initiaitive title and description and click update button, initiative title and description is changed. So i just test after update, they are different from previous content. Do we need to test what to expect after users click the update button?

File: tests/acceptance/editing-initiative-test.js (hunk 2 above)
Comment (oliverbarnes) [inline tests/acceptance/editing-initiative-test.js]: Ok I see now - testing the contents of the input boxes does makes sense then. You can use jquery for this: http://stackoverflow.com/questions/10507294/how-to-get-the-value-of-a-textarea-in-jquery

File: tests/acceptance/editing-initiative-test.js (hunk 3 above)
Comment (oliverbarnes) [inline tests/acceptance/editing-initiative-test.js]: Sounds like this should be the first scenario, success. And you _can_ check the outcome, the behavior is for the form be replaced by the newly created initiative, in the initiative view. Simply test for title and description there

**Pull request closed without merging**